docs/design/messaging/message-cookie.md

```diff
@@ -0,0 +1,95 @@
+# Message cookie
+
+## Concept
+The goal of this approach is to be able to route response into specific handler (the handler that have produced request)
+into specific service into specific application/storm-bolt.
+
+The message producer must include into the message info unambiguously describing the sender (into hierarchical manner) - 
+the message cookie object. Message recipient must produce response and copy message cookie from request. So the only 
```

Comment:
still not a big fan of naming this cookie:-p

Reply:
It have been lost somewhere during rebase commit split actions. I have regenerated it from puml source.

Reply:
Token:)